### ISSUE TITLE
Broken on Mercurial 3.2.3

### ISSUE THREAD

**Issue by sethfowler** [dmnd/hg-blametrail#1]
This seems like a nice extension, and I'd like to try it, but unfortunately it doesn't work for me on Mercurial 3.2.3.

```
○ > hg blame --trail 10 image/src/imgLoader.h --context=2 --pager off
lines 10±2:
50546: 42: #define imgLoader_h__
50546: 43:
83718: 44: #include "mozilla/Attributes.h"
203977: 11: #include "mozilla/Mutex.h"
83718: 45:

changeset:   83718:20550c970bd4
user:        Jeff Walden <jwalden@mit.edu>
date:        Mon Jan 02 14:23:41 2012 -0600
summary:     Bug 714572 - Fix nearly all clang warnings building image/.  r=bholley

diff --git a/image/src/imgLoader.h b/image/src/imgLoader.h
--- a/image/src/imgLoader.h
+++ b/image/src/imgLoader.h
@@ -36,16 +36,18 @@
36  * the provisions above, a recipient may use your version of this file under
37  * the terms of any one of the MPL, the GPL or the LGPL.
38  *
39  * ***** END LICENSE BLOCK ***** */
40
41 #ifndef imgLoader_h__
42 #define imgLoader_h__
43
=>+#include "mozilla/Attributes.h"
  +
44 #include "imgILoader.h"
45 #include "imgICache.h"
46 #include "nsWeakReference.h"
47 #include "nsIContentSniffer.h"
48 #include "nsRefPtrHashtable.h"
49 #include "nsExpirationTracker.h"
50 #include "nsAutoPtr.h"
51 #include "prtypes.h"
** Unknown exception encountered with possibly-broken third-party extension blametrail
** which supports versions unknown of Mercurial.
** Please disable blametrail and try your action again.
** If that fixes the bug please report it to the extension author.
** Python 2.7.6 (default, Sep  9 2014, 15:04:36) [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)]
** Mercurial Distributed SCM (version 3.2.3)
** Extensions loaded: color, graphlog, strip, mq, rebase, record, histedit, pager, progress, transplant, shelve, purge, blametrail, bzexport, bzpost, mozext, qbackout, qimportbz, crecord, trychooser
Traceback (most recent call last):
  File "/usr/local/bin/hg", line 43, in <module>
    mercurial.dispatch.run()
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 28, in run
    sys.exit((dispatch(request(sys.argv[1:])) or 0) & 255)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 71, in dispatch
    ret = _runcatch(req)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 140, in _runcatch
    return _dispatch(req)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 850, in _dispatch
    cmdpats, cmdoptions)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 611, in runcommand
    ret = _runcommand(ui, options, cmd, d)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/extensions.py", line 196, in wrap
    return wrapper(origfn, *args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/hgext/pager.py", line 158, in pagecmd
    return orig(ui, options, cmd, cmdfunc)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/extensions.py", line 196, in wrap
    return wrapper(origfn, *args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/hgext/color.py", line 485, in colorcmd
    return orig(ui_, opts, cmd, cmdfunc)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 941, in _runcommand
    return checkargs()
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 912, in checkargs
    return cmdfunc()
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/dispatch.py", line 847, in <lambda>
    d = lambda: util.checksignature(func)(ui, *args, **cmdoptions)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/util.py", line 677, in check
    return func(*args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/extensions.py", line 151, in wrap
    util.checksignature(origfn), *args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/util.py", line 677, in check
    return func(*args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/hgext/mq.py", line 3462, in mqcommand
    return orig(ui, repo, *args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/util.py", line 677, in check
    return func(*args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/extensions.py", line 151, in wrap
    util.checksignature(origfn), *args, **kwargs)
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/util.py", line 677, in check
    return func(*args, **kwargs)
  File "/Users/mfowler/.config/hg/blametrail/blametrail.py", line 106, in blame_trail
    show_hunk(ui, repo, *pats, patch=True, rev=[rev], follow=True, date=None, line=line)
  File "/Users/mfowler/.config/hg/blametrail/blametrail.py", line 324, in show_hunk
    for ctx in cmdutil.walkchangerevs(repo, matchfn, opts, prep):
  File "/usr/local/Cellar/mercurial/3.2.3/lib/python2.7/site-packages/mercurial/cmdutil.py", line 1629, in iterate
    prepare(ctx, fns)
  File "/Users/mfowler/.config/hg/blametrail/blametrail.py", line 322, in prep
    displayer.show(ctx, copies=copies, matchfn=revmatchfn)
  File "/Users/mfowler/.config/hg/blametrail/blametrail.py", line 163, in show
    self._show(ctx, copies, matchfn, props)
  File "/Users/mfowler/.config/hg/blametrail/blametrail.py", line 172, in _show
    self.diff(diffopts, prev, node, match=matchfn)
  File "/Users/mfowler/.config/hg/blametrail/blametrail.py", line 195, in diff
    lines_from = (int(m.group(1)), int(m.group(2)))
AttributeError: 'NoneType' object has no attribute 'group'
```

**Comment by dmnd** [comment]
Sorry for the late response. I don't use Mercurial anymore so I won't be updating this myself. If you get inspired to, feel free to fix it up and take over :smile:

**Comment by evnu** [comment]
This might have been fixed by pull request #2 as well. I do not have Mercurial 3.2.3 available and cannot test this myself right now.

**Comment by dmnd** [comment]
Closing this since #2 _probably_ obsoletes it. Thanks, @evnu!
